app/controllers/volunteers_controller.rb

```diff
@@ -4,10 +4,14 @@ class VolunteersController < ApplicationController
   before_action :authenticate_lead, except: %i[new create destroy]
 
   def index
-    @volunteers = @volunteer_role.volunteers.order(:created_at)
-    respond_to do |format|
-      format.html
-      format.csv { send_data @volunteers.to_csv, filename: "volunteers-#{@volunteer_role.name}-#{Date.today}.csv" }
+    if current_user.lead_for?(@volunteer_role)
```

Comment:
is that not what the before_action is doing?

Reply:
Revertedthis now